### ISSUE TITLE
[ENH] estimator model type tag

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#6669]
Idea: should we have a tag or multiple that categorize models by algorithm kind?

What I mean is not the scientific type - forecaster, classifier - but the "fitting principle" that the model uses. High-level categories would be "classical statistical", "machine learning (excluding neural networks)", "deep learning (excluding foundation models)", "foundation models (= deep learning trained on large corpus)".

Similar categorizations exist in classification models already ("distance based", "deep learning"), and pairwise transformers also have a tag specifying whether they are a distance (triangle equality) or kernel (positive semi-definiteness).

This kind of information may be particularly helpful in retrieval and estimator selection for the user, e.g., by using the estimator search function in the overview page: https://www.sktime.net/en/stable/estimator_overview.html

Opening up the discussion, @sktime/core-developers

**Comment by yarnabrina** [comment]
We can have that tag. If I am honest, not sure if it will be widely used by users in programming part, but I agree that this can definitely help to look at options combined with tag based filters in updated docs pages.

**Comment by jgyasu** [comment]
> This kind of information may be particularly helpful in retrieval and estimator selection for the user

I agree. And even from an estimator's developer perspective, it will be helpful for a developer to look for similar existing estimators in sktime while implementing an estimator.

I suppose this will involve adding the tag information in the templates for adding the estimator and then adding the tags to all the existing estimators? I do have a question though, let's say there's a forecaster like `TimeLLMForecaster`, it comes under "foundation models" but foundation models are themselves a subset of "deep learning" models so should the tag have both of them?

(This is just me thinking out loud but I think we can even seggregate the models on the sktime's website in sub-sections based on the tags)

**Comment by fkiraly** [comment]
> it comes under "foundation models" but foundation models are themselves a subset of "deep learning" models so should the tag have both of them?

Yes, that is exactly the question - what exactly the tag should be.

One solution would be to allow a set of tags in the field, that is, the tag is a set and it could be `["deep_learning", "foundation_model"]`. The `all_estimators` search already allows list-valued tags and searching for tags. One possible design, hence, is to have a single tag field with a list-valued tag field value.

Which does not solve the question though, what all the tags would be, even for forecasters it seems like a hard question.

There is some taxonomy in the API reference that we could start from.

**Comment by gbilleyPeco** [comment]
I think allowing a list of tags is a good idea. There should also be a tag for if the model is pre-trained, vs if the implementation is just the model architecture. (Probably only relevant for foundation models)

**Comment by jgyasu** [comment]
> There is some taxonomy in the API reference that we could start from.

@fkiraly I did not get this part, can you please explain this a bit? We can also open a discussion thread on Discord to allow people to comment on the set of exhaustive tags for the estimators in sktime.

**Comment by fkiraly** [comment]
I meant, if you look at the headers on this page:
https://www.sktime.net/en/latest/api_reference/forecasting.html

they already categorize the estimators in a way, and can be seen as a kind of taxonomy

**Comment by mabuimo** [comment]
My tags proposal is aligned with Franz´s initial message: 
- Statistical
- ML
- DL
- FM

**Comment by fkiraly** [comment]
so, `info:model_type` as a tag name?

**Comment by jgyasu** [comment]
> so, info:model_type as a tag name?

Sounds good! I suppose we have decided to have multiple categories (e.g., DL, FM) for some models, that means the tag is a set. Does the current tag system handles a list as a key?

**Comment by fkiraly** [comment]
> Does the current tag system handles a list as a key?

Yes, `all_objects` is compatible with list-valued tags, and treats them as logical conjunction by default, i.e., "and". So a list `[tagvalue1, tagvalue 2]` is interpreted as the object having value `tagvalue1` *and* `tagvalue2`.

**Comment by mabuimo** [comment]
I would argue that FMs shouldn't also be in the DL category to properly differentiate them.
What would be of interest for FMs is to add a tag to distinguish between zero-shot and few-shot capability.

**Comment by fkiraly** [comment]
hmmmm, but FM are *also* DL? Why should it not have both tags?

**Comment by mabuimo** [comment]
I just wanted to simplify. However, it is not a big deal. If someone wants to filter DL models which are not FM, it can concatenate the tags with FM:false. We are aligned.
